scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,68 +116,167 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
+                PackageDependencyScope.DEFAULT);
+        List<Package> directDependencies = packageResolution.dependencyGraph().getDirectDependencies(rootPkgNode)
+                .stream().map(ResolvedPackageDependency::packageInstance).toList();
+        Map<String, List<Rule>> externalAnalyzers = new HashMap<>();
+        for (Package pkgDependency : directDependencies) {
+            PackageManifest pkgManifest = pkgDependency.manifest();
+            String org = pkgManifest.org().value();
+            String name = pkgManifest.name().value();
+            String pluginName = org + FORWARD_SLASH + name;
+            if (pkgManifest.compilerPluginDescriptor().isEmpty() || !analyzerDescriptors.contains(pluginName)) {
+                continue;
+            }
 
-        // TODO: Implement the logic to pass scanner contexts to compiler plugins and perform external static analysis.
+            CompilerPluginDescriptor pluginDesc = pkgManifest.compilerPluginDescriptor().get();
+            List<URL> jarUrls = pluginDesc.dependencies().stream().map(dependency -> {
+                        try {
+                            return Path.of(dependency.getPath()).toUri().toURL();
+                        } catch (MalformedURLException ex) {
+                            throw new RuntimeException(ex);
+                        }
+            }).toList();
 
-        List<Issue> externalIssues = new ArrayList<>(scannerContextList.size());
-        for (ScannerContext scannerContext : scannerContextList) {
-            ReporterImpl reporter = (ReporterImpl) scannerContext.getReporter();
-            externalIssues.addAll(reporter.getIssues());
+            URLClassLoader ucl = URLClassLoader.newInstance(jarUrls.toArray(new URL[0]),
+                    this.getClass().getClassLoader());
+            InputStream resourceAsStream = ucl.getResourceAsStream(RULES_FILE);
+            if (resourceAsStream == null) {
+                continue;
+            }
+
+            String output;
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(resourceAsStream,
+                    StandardCharsets.UTF_8))) {
+                output = reader.lines().collect(Collectors.joining());
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            }
+
+            String fqn = pluginDesc.plugin().getClassName();
+            List<Rule> externalRules = new ArrayList<>();
+            Gson gson = new GsonBuilder().setPrettyPrinting().create();
+            JsonElement element = gson.fromJson(output, JsonElement.class);
+            if (!element.isJsonArray()) {
+                outputStream.println("Invalid JSON format for " + RULES_FILE + " in plugin '" + pluginName + "'.");
+                outputStream.println("Expected an array, but found:");
+                outputStream.println(gson.toJson(element));
+                return Optional.empty();
```

Comment:
1. Shall we move this into a separate method and pass these messages as parameters? this method can be used for the L192 - L195 as well.
2. Make these String a constant